pdfsam-basic: UncaughtExceptionLogger when Starting Application

### Nixpkgs version

- Stable (25.05)

### Describe the bug

## Information
Error is produced when starting up the application. A "pdfsam" banner briefly appears before crashing.

## Additional Information
Host information:
```
          ▗▄▄▄       ▗▄▄▄▄    ▄▄▄▖             marcel@marcel-pc
          ▜███▙       ▜███▙  ▟███▛             ----------------
           ▜███▙       ▜███▙▟███▛              OS: NixOS 25.05 (Warbler) x86_64
            ▜███▙       ▜██████▛               Host: B550 AORUS ELITE
     ▟█████████████████▙ ▜████▛     ▟▙         Kernel: Linux 6.16.7-zen1
    ▟███████████████████▙ ▜███▙    ▟██▙        Uptime: 9 hours, 40 mins
           ▄▄▄▄▖           ▜███▙  ▟███▛        Packages: 6434 (nix-system)
          ▟███▛             ▜██▛ ▟███▛         Shell: bash 5.2.37
         ▟███▛               ▜▛ ▟███▛          Display (DELL G3223Q): 3840x2160 @ 144 Hz (as 3072x1728) in 32" [External] *
▟███████████▛                  ▟██████████▙    Display (AORUS FI27Q): 1440x2560 @ 144 Hz (as 1253x2229) in 27" [External]
▜██████████▛                  ▟███████████▛    DE: KDE Plasma
      ▟███▛ ▟▙               ▟███▛             WM: KWin (Wayland)
     ▟███▛ ▟██▙             ▟███▛              WM Theme: Breeze
    ▟███▛  ▜███▙           ▝▀▀▀▀               Theme: Breeze (Dark) [Qt]
    ▜██▛    ▜███▙ ▜██████████████████▛         Icons: breeze-dark [Qt], breeze-dark [GTK2/3/4]
     ▜▛     ▟████▙ ▜████████████████▛          Font: Noto Sans (10pt) [Qt], Noto Sans (10pt) [GTK2/3/4]
           ▟██████▙       ▜███▙                Cursor: breeze (24px)
          ▟███▛▜███▙       ▜███▙               Terminal: kitty 0.42.1
         ▟███▛  ▜███▙       ▜███▙              Terminal Font: Hack-Regular (11pt)
         ▝▀▀▀    ▀▀▀▀▘       ▀▀▀▘              CPU: AMD Ryzen 5 5600X (12) @ 4.65 GHz
                                               GPU: NVIDIA GeForce RTX 3060 Ti [Discrete]
                                               Memory: 10.85 GiB / 31.25 GiB (35%)
                                               Swap: Disabled
                                               Disk (/): 1.62 TiB / 3.58 TiB (45%) - ext4
                                               Local IP (eno1): 192.168.1.25/24
                                               Locale: en_GB.UTF-8
```

### Steps to reproduce

1) Pull pdfsam from package manager.
2) Run pdfsam via terminal.
3) Error is produced on startup.

### Expected behaviour

The app launches as expected

### Screenshots

_No response_

### Relevant log output

```console
INFO 20:48:05.912 org.pdfsam.gui.PdfsamApp.init [JavaFX-Launcher] Starting...
INFO 20:48:06.198 org.sejda.core.context.DefaultSejdaConfiguration.<init> [JavaFX Application Thread] Configuring Sejda 5.1.10
Exception in Application start method
INFO 20:48:06.519 org.pdfsam.gui.PdfsamApp.stop [JavaFX Application Thread] Closing PDFsam...
ERROR 20:48:06.543 org.pdfsam.gui.UncaughtExceptionLogger.uncaughtException [main] Unexpected error: java.lang.RuntimeException: Exception in Application start method
	at javafx.graphics/com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:893)
	at javafx.graphics/com.sun.javafx.application.LauncherImpl.lambda$launchApplication$2(LauncherImpl.java:196)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.NoSuchMethodError: 'javafx.application.Platform$Preferences javafx.application.Platform.getPreferences()'
	at org.pdfsam.gui@5.3.1/org.pdfsam.gui.theme.Themes.defaultTheme(Themes.java:66)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at org.pdfsam.gui@5.3.1/org.pdfsam.gui.theme.Themes.getOrDefault(Themes.java:50)
	at org.pdfsam.gui@5.3.1/org.pdfsam.gui.PdfsamApp.initTheme(PdfsamApp.java:184)
	at org.pdfsam.gui@5.3.1/org.pdfsam.gui.PdfsamApp.initScene(PdfsamApp.java:173)
	at org.pdfsam.gui@5.3.1/org.pdfsam.gui.PdfsamApp.start(PdfsamApp.java:116)
	at javafx.graphics/com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$9(LauncherImpl.java:839)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runAndWait$12(PlatformImpl.java:483)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:456)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:455)
	at javafx.graphics/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
	at javafx.graphics/com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at javafx.graphics/com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$10(GtkApplication.java:263)
	... 1 morejava.lang.RuntimeException: Exception in Application start method
	at javafx.graphics/com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:893)
	at javafx.graphics/com.sun.javafx.application.LauncherImpl.lambda$launchApplication$2(LauncherImpl.java:196)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.NoSuchMethodError: 'javafx.application.Platform$Preferences javafx.application.Platform.getPreferences()'
	at org.pdfsam.gui@5.3.1/org.pdfsam.gui.theme.Themes.defaultTheme(Themes.java:66)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at org.pdfsam.gui@5.3.1/org.pdfsam.gui.theme.Themes.getOrDefault(Themes.java:50)
	at org.pdfsam.gui@5.3.1/org.pdfsam.gui.PdfsamApp.initTheme(PdfsamApp.java:184)
	at org.pdfsam.gui@5.3.1/org.pdfsam.gui.PdfsamApp.initScene(PdfsamApp.java:173)
	at org.pdfsam.gui@5.3.1/org.pdfsam.gui.PdfsamApp.start(PdfsamApp.java:116)
	at javafx.graphics/com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$9(LauncherImpl.java:839)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runAndWait$12(PlatformImpl.java:483)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:456)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:455)
	at javafx.graphics/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
	at javafx.graphics/com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at javafx.graphics/com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$10(GtkApplication.java:263)
	... 1 more
```

### Additional context

The upstream issue (https://github.com/torakiki/pdfsam/issues/785) maintainers mention this is likely a missing JavaFX problem.

### System metadata

TBD (will fill out today evening)

### Notify maintainers

@1000101

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/misc/pdfsam-basic/default.nix
+++ b/pkgs/applications/misc/pdfsam-basic/default.nix
@@ -3,63 +3,95 @@
   stdenv,
   makeDesktopItem,
   fetchurl,
-  jdk21,
+  temurin-jre-bin-21,
+  javaPackages,
   wrapGAppsHook3,
-  glib,
+  dpkg,
+  xorg,
+  gtk3,
+  libGL,
+  alsa-lib,
   nix-update-script,
+  desktop-file-utils,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "pdfsam-basic";
-  version = "5.3.2";
+  version = "5.4.1";
 
   src = fetchurl {
-    url = "https://github.com/torakiki/pdfsam/releases/download/v${version}/pdfsam-basic_${version}-1_amd64.deb";
-    hash = "sha256-Y0Q9uT6cyxIYTX0JxoS0r3TamPT1iLXr94Zex30AeWo=";
+    url = "https://github.com/torakiki/pdfsam/releases/download/v${finalAttrs.version}/pdfsam-basic_${finalAttrs.version}-1_amd64.deb";
+    hash = "sha256-iM0avC0YwxaB2prWbiKJZ9Fzd/HcdDWJg5IWRmNlVkM=";
   };
 
-  unpackPhase = ''
-    ar vx ${src}
-    tar xvf data.tar.gz
-  '';
-
-  nativeBuildInputs = [ wrapGAppsHook3 ];
-  buildInputs = [ glib ];
-
-  preFixup = ''
-    gappsWrapperArgs+=(--set JAVA_HOME "${jdk21}" --set PDFSAM_JAVA_PATH "${jdk21}")
-  '';
+  nativeBuildInputs = [
+    dpkg
+    wrapGAppsHook3
+    desktop-file-utils
+  ];
 
   installPhase = ''
-    cp -R opt/pdfsam-basic/ $out/
-    mkdir -p "$out"/share/icons
-    cp --recursive ${desktopItem}/share/applications $out/share
-    cp $out/icon.svg "$out"/share/icons/pdfsam-basic.svg
+    runHook preInstall
+    desktop-file-edit usr/share/applications/pdfsam-basic.desktop \
+         --set-key="Exec" --set-value="pdfsam-basic %F" \
+         --set-key="Path" --set-value="$out/share/pdfsam-basic" \
+         --set-icon="pdfsam-basic"
+       mkdir $out
+       cp -r usr/share $out/share
+       mkdir $out/share/pdfsam-basic
+       cp -r opt/pdfsam-basic/lib $out/share/pdfsam-basic/lib
+       install -Dm0644 opt/pdfsam-basic/splash.png $out/share/pdfsam-basic/splash.png
+       install -Dm0644 opt/pdfsam-basic/icon.svg $out/share/icons/hicolor/scalable/apps/pdfsam-basic.svg
+       mkdir $out/bin
+       makeWrapper ${temurin-jre-bin-21}/bin/java $out/bin/pdfsam-basic \
+         "''${gappsWrapperArgs[@]}" \
+         --set JAVA_HOME ${temurin-jre-bin-21} \
+         --set PDFSAM_JAVA_PATH ${temurin-jre-bin-21} \
+         --prefix LD_LIBRARY_PATH : ${
+           lib.makeLibraryPath [
+             javaPackages.openjfx23 # PDFSam Basic requires JDK 21 and JavaFX 23 https://github.com/torakiki/pdfsam/issues/785#issuecomment-3446564717
+             xorg.libXxf86vm
+             xorg.libXtst
+             gtk3
+             libGL
+             alsa-lib
+           ]
+         } \
+         --add-flags ${
+           lib.escapeShellArg (
+             lib.escapeShellArgs [
+               "--enable-preview"
+               "--module-path"
+               "${placeholder "out"}/share/pdfsam-basic/lib"
+               "--module"
+               "org.pdfsam.basic/org.pdfsam.basic.App"
+               "-Xmx512M"
+               "-splash:${placeholder "out"}/share/pdfsam-basic/splash.png"
+               "-Dapp.name=\"pdfsam-basic\""
+               "-Dapp.pid=\"$$\""
+               "-Dapp.home=\"${placeholder "out"}/share/pdfsam-basic\""
+               "-Dbasedir=\"${placeholder "out"}/share/pdfsam-basic\""
+               "-Dprism.lcdtext=false"
+             ]
+           )
+         }
+    runHook postInstall
   '';
 
-  desktopItem = makeDesktopItem {
-    name = pname;
-    exec = pname;
-    icon = pname;
-    comment = meta.description;
-    desktopName = "PDFsam Basic";
-    genericName = "PDF Split and Merge";
-    mimeTypes = [ "application/pdf" ];
-    categories = [ "Office" ];
-  };
+  dontWrapGApps = true;
 
   passthru.updateScript = nix-update-script { };
 
-  meta = with lib; {
+  meta = {
     homepage = "https://github.com/torakiki/pdfsam";
     description = "Multi-platform software designed to extract pages, split, merge, mix and rotate PDF files";
     mainProgram = "pdfsam-basic";
-    sourceProvenance = with sourceTypes; [
+    sourceProvenance = with lib.sourceTypes; [
       binaryBytecode
       binaryNativeCode
     ];
-    license = licenses.agpl3Plus;
+    license = lib.licenses.agpl3Plus;
     platforms = [ "x86_64-linux" ];
-    maintainers = with maintainers; [ _1000101 ];
+    maintainers = with lib.maintainers; [ _1000101 ];
   };
-}
+})

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -11922,9 +11922,7 @@ with pkgs;
     lua = lua5;
   };
 
-  pdfsam-basic = callPackage ../applications/misc/pdfsam-basic {
-    jdk21 = openjdk21.override { enableJavaFX = true; };
-  };
+  pdfsam-basic = callPackage ../applications/misc/pdfsam-basic { };
 
   mupdf-headless = mupdf.override {
     enableX11 = false;

